### ISSUE TITLE
Plants as treatment results with nitrogen in the path

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/Feedback#433]
I'm seeing a lot of vegetables in results from ARAGORN.  It looks like its when nitrogen is in the path:

![image](https://github.com/NCATSTranslator/Feedback/assets/38321826/4a425bd6-7cc1-4b95-8f6a-b31b7ef5f51f)

Sneddon Syndrome:  https://ui.test.transltr.io/results?l=Sneddon%20Syndrome&i=MONDO:0008436&t=0&q=2b273528-c50f-4070-925d-bec92a99eaba

Bethlem Myopathy:  https://ui.test.transltr.io/results?l=Bethlem%20Myopathy&i=MONDO:0008029&t=0&q=f9f6b232-d754-4830-886f-8436889b6529

looks like through tyrosine for Alzheimer:  https://ui.test.transltr.io/results?l=Alzheimer%20Disease&i=MONDO:0004975&t=0&q=ea463b48-d662-4f5c-bf6b-638deb2b89bb

![image](https://github.com/NCATSTranslator/Feedback/assets/38321826/1afc1392-1369-44b2-a6dd-cb5b1d5e8b74)

**Comment by sierra-moxon** [comment]
@cbizon - retested on CI this morning and I am not seeing `Parsley` as a result any longer.  
https://ui.ci.transltr.io/results?l=Sneddon%20Syndrome&i=MONDO:0008436&t=0&q=cfbb36b0-1bef-47e7-8103-ebf41a101611

I do see some interesting results as treatments for Sneddon Syndrome including:
`Chlorophyll`
`Nitrogen mustard` (the chemical warfare agent)
`Nitrogen 50 %`

but I think these must be from SEMMED?  

<img width="1286" alt="Screen Shot 2023-08-09 at 10 46 40 AM" src="https://github.com/NCATSTranslator/Feedback/assets/4561292/6ed91e12-68fe-4053-8251-cda9798b22f4">

<img width="1287" alt="Screen Shot 2023-08-09 at 10 47 02 AM" src="https://github.com/NCATSTranslator/Feedback/assets/4561292/530721b2-c9ca-4720-af4c-1a1501b09401">

**Comment by andrewsu** [comment]
**TLDR: None of the edges in these answers references semmeddb as far as I can see, so I'm not sure where the UI is getting semmeddb as a source.  I also spot checked nitrogen mustard and that appears to be the same.  Going to tag @cbizon since this is an Aragorn answer and maybe I'm missing something, but as far as I can see it seems to be a UI issue so going to reassign to @Genomewide.**

More details below...

Looking into the chlorophyll (`PUBCHEM.COMPOUND:16667503`) answer, this one is from aragorn, result # 469.
```
            {
                "analyses": [
                    {
                        "edge_bindings": {
                            "t_edge": [
                                {
                                    "id": "ef0e6dd3-c04b-4339-8f0a-de3a6a659722"
                                }
                            ]
                        },
                        "resource_id": "infores:aragorn",
                        "score": 0.532907792721096,
                        "support_graphs": [
                            "OMNICORP_support_graph_651"
                        ]
                    }
                ],
                "node_bindings": {
                    "on": [
                        {
                            "id": "MONDO:0008436"
                        }
                    ],
                    "sn": [
                        {
                            "id": "PUBCHEM.COMPOUND:16667503"
                        }
                    ]
                },
                "normalized_score": 4.4,
                "resource_id": "Aragorn"
            },
```

the support graph has two edges:

```
            "OMNICORP_support_graph_651": {
                "edges": [
                    "9c25f696-8908-4e72-856c-bb6336d7b110",
                    "5a677c53-3bf6-4f95-8ccd-0a153ed9bef3"
                ]
            },
```
This edge says `Bosentan - occurs_together_in_literature_with - Nitrogen`:
```
                "9c25f696-8908-4e72-856c-bb6336d7b110": {
                    "attributes": [
                        {
                            "attribute_type_id": "biolink:has_count",
                            "original_attribute_name": "num_publications",
                            "value": 5,
                            "value_type_id": "EDAM:data_0006"
                        },
                        {
                            "attribute_type_id": "biolink:has_numeric_value",
                            "original_attribute_name": "weight",
                            "value": 1,
                            "value_type_id": "EDAM:data_1669"
                        }
                    ],
                    "object": "PUBCHEM.COMPOUND:947",
                    "predicate": "biolink:occurs_together_in_literature_with",
                    "sources": [
                        {
                            "resource_id": "infores:omnicorp",
                            "resource_role": "primary_knowledge_source"
                        },
                        {
                            "resource_id": "infores:ara-aragorn",
                            "resource_role": "aggregator_knowledge_source",
                            "source_record_urls": null,
                            "upstream_resource_ids": null
                        }
                    ],
                    "subject": "PUBCHEM.COMPOUND:104865"
                },
```
This edge says `Chlorophyll - occurs_together_in_literature_with - Nitrogen`:
```
                "5a677c53-3bf6-4f95-8ccd-0a153ed9bef3": {
                    "attributes": [
                        {
                            "attribute_type_id": "biolink:has_count",
                            "original_attribute_name": "num_publications",
                            "value": 3423,
                            "value_type_id": "EDAM:data_0006"
                        },
                        {
                            "attribute_type_id": "biolink:has_numeric_value",
                            "original_attribute_name": "weight",
                            "value": 1,
                            "value_type_id": "EDAM:data_1669"
                        }
                    ],
                    "object": "PUBCHEM.COMPOUND:947",
                    "predicate": "biolink:occurs_together_in_literature_with",
                    "sources": [
                        {
                            "resource_id": "infores:omnicorp",
                            "resource_role": "primary_knowledge_source"
                        },
                        {
                            "resource_id": "infores:ara-aragorn",
                            "resource_role": "aggregator_knowledge_source",
                            "source_record_urls": null,
                            "upstream_resource_ids": null
                        }
                    ],
                    "subject": "PUBCHEM.COMPOUND:16667503"
                },
```
All of this matches what I see in the ARAX UI (https://arax.ci.transltr.io/?r=9e00e1a8-e741-4d7d-ac78-985c311eaf6e):

![image](https://github.com/NCATSTranslator/Feedback/assets/2635409/c06ba507-3b04-4f29-95e1-f0426d725fff)

Notably, none of the edges above reference semmeddb, so I'm not sure where the UI is getting semmeddb as a source.

**Comment by cbizon** [comment]
@andrewsu I think that you're misunderstanding the `analysis.support_graph` element.  That support_graph is not the graph giving the path that imply the edge.  Instead it is giving the paths that support the analysis (i.e. the score). The literature co-occurrence are only used in coming up with the numerical score value.

Instead you want to find `ef0e6dd3-c04b-4339-8f0a-de3a6a659722` in the edges and look for its attributes, and in there you'll find the support graph pointing to the edges showing up in the UI.  One of those (Bosentan treats Sneddon Syndrome) is coming from semmeddb and pointing at https://pubmed.ncbi.nlm.nih.gov/26000141/ which seems reasonable.

So the real problem on this one (IMO) is that ARAGORN is still allowing a low information content node (Nitrogen) in a path leading to a bad answer.  I believe that the new nodenorm will use the more careful information content, which lowers the value of Nitrogen below Aragorn's threshold. Tagging @maximusunc to keep an eye on it.

**Comment by andrewsu** [comment]
ahh, thanks @cbizon for the correction! and I also see now where this can be viewed within the ARAX UI. I think that wasn't the first time I missed the _other_ location for support graphs, unfortunately...  😳

So just to follow up on this from the semmeddb perspective, it looks like there are two edges in the supporting graph that reference semmeddb -- one that has 1 PMID in support, and one that has 2 PMIDs in support.  Both of these edges will be filtered out under https://github.com/NCATSTranslator/Feedback/issues/100, which restricts semmeddb to edges with a pmid_count > 3.  so **chlorophyll** should disappear with the 8/18 code freeze.

![image](https://github.com/NCATSTranslator/Feedback/assets/2635409/e6bf4d82-7bce-44b1-9f8f-f14e4a3d2f29)

**Nitrogen mustard** does not depend on semmeddb, so it will not be affected by semmeddb changes.

**Nitrogen 50 %** uses one of the same semmeddb edges as for **chlorophyll**, so this answer should disappear with the 8/18 code freeze.

**Comment by maximusunc** [comment]
I just ran these three queries again today and confirmed that those results are no longer present:

Sneddon Syndrome: https://ui.test.transltr.io/main/results?l=Sneddon%20Syndrome&i=MONDO:0008436&t=0&q=e0462ae6-9b72-4a6d-9b01-77742ef129cf

Bethlem Myopathy: https://ui.test.transltr.io/main/results?l=Bethlem%20Myopathy&i=MONDO:0008029&t=0&q=8d94f75c-08c4-4c3c-8482-3f3d89a850d2

Alzheimer: https://ui.test.transltr.io/main/results?l=Alzheimer%20Disease&i=MONDO:0004975&t=0&q=3c82a131-6ae0-4bf4-8ba7-8f9519d95569

Should we close this issue?
